### ISSUE TITLE
segfault on mirroring displays

### ISSUE THREAD

**Issue by TRPB** [WayfireWM/wayfire#1408]
**Describe the bug**

When enabling the mirror mode, wayfire segfaults. 

I'm not sure how linearly wayfire parses the configuration file, but I made sure to configure the display being mirrored before the one displaying the mirror. 

It shouldn't make any difference but `HDMI-A-1` is a capture card and not a normal display. 

I thought I could just set the position to `3840,0` to both displays to have them mirrored but this actually creates two different outputs on top of each other at the same position. You can only open windows on one, not both.

**To Reproduce**

My `wayfire.ini` looks like this:

```
[output:DP-3]
mode = 3840x2160@60000
position = 0,0
transform = normal
scale = 1.0000

[output:DP-2]
mode = 3840x2160@60000
position = 3840,0
transform = normal
scale = 1.000

[output:HDMI-A-1]
mode = mirror DP-2
```

**Expected behavior**

It should mirror the existing display correctly.

**Screenshots or stacktrace**

```
II 14-01-22 18:23:11.998 - [backend/drm/drm.c:1422] connector DP-2: Requesting modeset
II 14-01-22 18:23:11.998 - [src/core/output-layout.cpp:1068] new output: DP-2 ("Acer Technologies Acer CB241HQK 0x0000F458")
II 14-01-22 18:23:11.998 - [src/core/output-layout.cpp:428] loaded mode 3840x2160@60000
II 14-01-22 18:23:12.007 - [backend/drm/drm.c:717] connector DP-2: Modesetting with '3840x2160@59997mHz'
type equals variant: [type: string, value: toplevel]
(!title contains variant: [type: string, value: Ulauncher])
(type equals variant: [type: string, value: toplevel] | (type equals variant: [type: string, value: x-or] & focusable equals variant: [type: bool, value: 1]))
type equals variant: [type: string, value: overlay]
false
false
title contains variant: [type: string, value: Ulauncher]
II 14-01-22 18:23:12.085 - [backend/drm/drm.c:1422] connector DP-3: Requesting modeset
II 14-01-22 18:23:12.085 - [src/core/output-layout.cpp:1068] new output: DP-3 ("Acer Technologies Acer CB241HQK 0x0000F457")
II 14-01-22 18:23:12.085 - [src/core/output-layout.cpp:428] loaded mode 3840x2160@60000
II 14-01-22 18:23:12.085 - [src/core/output-layout.cpp:428] loaded mode 3840x2160@60000
II 14-01-22 18:23:12.089 - [backend/drm/drm.c:717] connector DP-3: Modesetting with '3840x2160@59997mHz'
type equals variant: [type: string, value: toplevel]
(!title contains variant: [type: string, value: Ulauncher])
(type equals variant: [type: string, value: toplevel] | (type equals variant: [type: string, value: x-or] & focusable equals variant: [type: bool, value: 1]))
type equals variant: [type: string, value: overlay]
false
false
title contains variant: [type: string, value: Ulauncher]
II 14-01-22 18:23:12.108 - [backend/drm/drm.c:1422] connector HDMI-A-1: Requesting modeset
II 14-01-22 18:23:12.108 - [src/core/output-layout.cpp:1068] new output: HDMI-A-1 ("Unknown HD60 S+ 0x00000000")
II 14-01-22 18:23:12.108 - [src/core/output-layout.cpp:428] loaded mode 3840x2160@60000
II 14-01-22 18:23:12.108 - [src/core/output-layout.cpp:428] loaded mode mirror DP-2
II 14-01-22 18:23:12.108 - [src/core/output-layout.cpp:428] loaded mode 3840x2160@60000
II 14-01-22 18:23:12.110 - [backend/drm/drm.c:717] connector HDMI-A-1: Modesetting with '3840x2160@60000mHz'
EE 14-01-22 18:23:12.118 - [types/wlr_cursor.c:856] Cannot map device "Power Button" to output(not found in this cursor)
EE 14-01-22 18:23:12.118 - [types/wlr_cursor.c:856] Cannot map device "Power Button" to output(not found in this cursor)
EE 14-01-22 18:23:12.118 - [types/wlr_cursor.c:856] Cannot map device "Logitech G502" to output(not found in this cursor)
EE 14-01-22 18:23:12.118 - [types/wlr_cursor.c:856] Cannot map device "ROCCAT ROCCAT Ryos MK FX" to output(not found in this cursor)
EE 14-01-22 18:23:12.118 - [types/wlr_cursor.c:856] Cannot map device "ROCCAT ROCCAT Ryos MK FX Consumer Control" to output(not found in this cursor)
EE 14-01-22 18:23:12.118 - [types/wlr_cursor.c:856] Cannot map device "ROCCAT ROCCAT Ryos MK FX System Control" to output(not found in this cursor)
EE 14-01-22 18:23:12.118 - [types/wlr_cursor.c:856] Cannot map device "ROCCAT ROCCAT Ryos MK FX" to output(not found in this cursor)
EE 14-01-22 18:23:12.118 - [types/wlr_cursor.c:856] Cannot map device "Blue Microphones Yeti Nano Consumer Control" to output(not found in this cursor)
EE 14-01-22 18:23:12.118 - [types/wlr_cursor.c:856] Cannot map device "BRIO 4K Stream Edition" to output(not found in this cursor)
EE 14-01-22 18:23:12.118 - [types/wlr_cursor.c:856] Cannot map device "BRIO 4K Stream Edition Consumer Control" to output(not found in this cursor)
EE 14-01-22 18:23:12.118 - [types/wlr_cursor.c:856] Cannot map device "Game Capture HD60 S+: Game Capt" to output(not found in this cursor)
EE 14-01-22 18:23:12.135 - [EGL] command: eglCreateImageKHR, error: EGL_BAD_ALLOC (0x3003), message: "dri2_create_image_khr_texture"
EE 14-01-22 18:23:12.135 - [render/egl.c:681] eglCreateImageKHR failed
EE 14-01-22 18:23:12.135 - [render/gles2/texture.c:271] Failed to create EGL image from DMA-BUF
EE 14-01-22 18:23:12.135 - [src/main.cpp:132] Fatal error: Segmentation fault
EE 14-01-22 18:23:12.164 - #1  wf::print_trace(bool) ??:?
EE 14-01-22 18:23:12.185 - #2  __restore_rt sigaction.c:?
EE 14-01-22 18:23:12.206 - #3  wlr_texture_is_gles2 ??:?
EE 14-01-22 18:23:12.231 - #4  wf::texture_t::texture_t(wlr_texture*) ??:?
EE 14-01-22 18:23:12.256 - #5  wf::output_layout_t::get_output_coords_at(wf::pointf_t, wf::pointf_t&) ??:?
EE 14-01-22 18:23:12.281 - #6  _start ??:?
EE 14-01-22 18:23:12.301 - #7  wlr_signal_emit_safe ??:?
EE 14-01-22 18:23:12.321 - #8  wlr_backend_finish ??:?
EE 14-01-22 18:23:12.341 - #9  drmHandleEvent ??:?
EE 14-01-22 18:23:12.361 - #10 wlr_drm_connector_get_panel_orientation ??:?
EE 14-01-22 18:23:12.381 - #11 wl_event_loop_dispatch ??:?
EE 14-01-22 18:23:12.401 - #12 wl_display_run ??:?
EE 14-01-22 18:23:12.426 - #13 main ??:?
EE 14-01-22 18:23:12.449 - #14 __libc_start_main ??:?
EE 14-01-22 18:23:12.473 - #15 _start ??:?
```

**Wayfire version**

git master

I'm using a radeon VII and here's my wlr-randr output:

```
DP-2 "Acer Technologies Acer CB241HQK 0x0000F458 (DP-2)"
  Physical size: 520x290 mm
  Enabled: yes
  Modes:
    720x400 px, 70.082001 Hz
    640x480 px, 59.939999 Hz
    640x480 px, 59.939999 Hz
    640x480 px, 60.000000 Hz
    640x480 px, 66.667000 Hz
    640x480 px, 72.808998 Hz
    640x480 px, 75.000000 Hz
    720x480 px, 59.939999 Hz
    720x480 px, 59.939999 Hz
    720x480 px, 60.000000 Hz
    720x480 px, 60.000000 Hz
    720x576 px, 50.000000 Hz
    720x576 px, 50.000000 Hz
    800x600 px, 56.250000 Hz
    800x600 px, 60.317001 Hz
    800x600 px, 72.188004 Hz
    800x600 px, 75.000000 Hz
    832x624 px, 74.551003 Hz
    1024x768 px, 60.004002 Hz
    1024x768 px, 70.069000 Hz
    1024x768 px, 75.028999 Hz
    1280x720 px, 50.000000 Hz
    1280x720 px, 59.939999 Hz
    1280x720 px, 60.000000 Hz
    1280x720 px, 60.000000 Hz
    1280x768 px, 59.869999 Hz
    1152x864 px, 75.000000 Hz
    1280x800 px, 59.810001 Hz
    1280x960 px, 60.000000 Hz
    1440x900 px, 59.887001 Hz
    1280x1024 px, 60.020000 Hz
    1280x1024 px, 75.025002 Hz
    1680x1050 px, 59.953999 Hz
    1600x1200 px, 59.997002 Hz
    1920x1080 px, 23.976000 Hz
    1920x1080 px, 24.000000 Hz
    1920x1080 px, 29.969999 Hz
    1920x1080 px, 30.000000 Hz
    1920x1080 px, 50.000000 Hz
    1920x1080 px, 59.939999 Hz
    1920x1080 px, 60.000000 Hz
    1920x1080 px, 60.000000 Hz
    1920x1200 px, 59.997002 Hz
    3840x2160 px, 23.976000 Hz
    3840x2160 px, 24.000000 Hz
    3840x2160 px, 29.969999 Hz
    3840x2160 px, 30.000000 Hz
    3840x2160 px, 30.000000 Hz
    3840x2160 px, 59.997002 Hz (preferred, current)
  Position: 3840,0
  Transform: normal
  Scale: 1.000000
DP-3 "Acer Technologies Acer CB241HQK 0x0000F457 (DP-3)"
  Physical size: 520x290 mm
  Enabled: yes
  Modes:
    720x400 px, 70.082001 Hz
    640x480 px, 59.939999 Hz
    640x480 px, 59.939999 Hz
    640x480 px, 60.000000 Hz
    640x480 px, 66.667000 Hz
    640x480 px, 72.808998 Hz
    640x480 px, 75.000000 Hz
    720x480 px, 59.939999 Hz
    720x480 px, 59.939999 Hz
    720x480 px, 60.000000 Hz
    720x480 px, 60.000000 Hz
    720x576 px, 50.000000 Hz
    720x576 px, 50.000000 Hz
    800x600 px, 56.250000 Hz
    800x600 px, 60.317001 Hz
    800x600 px, 72.188004 Hz
    800x600 px, 75.000000 Hz
    832x624 px, 74.551003 Hz
    1024x768 px, 60.004002 Hz
    1024x768 px, 70.069000 Hz
    1024x768 px, 75.028999 Hz
    1280x720 px, 50.000000 Hz
    1280x720 px, 59.939999 Hz
    1280x720 px, 60.000000 Hz
    1280x720 px, 60.000000 Hz
    1280x768 px, 59.869999 Hz
    1152x864 px, 75.000000 Hz
    1280x800 px, 59.810001 Hz
    1280x960 px, 60.000000 Hz
    1440x900 px, 59.887001 Hz
    1280x1024 px, 60.020000 Hz
    1280x1024 px, 75.025002 Hz
    1680x1050 px, 59.953999 Hz
    1600x1200 px, 59.997002 Hz
    1920x1080 px, 23.976000 Hz
    1920x1080 px, 24.000000 Hz
    1920x1080 px, 29.969999 Hz
    1920x1080 px, 30.000000 Hz
    1920x1080 px, 50.000000 Hz
    1920x1080 px, 59.939999 Hz
    1920x1080 px, 60.000000 Hz
    1920x1080 px, 60.000000 Hz
    1920x1200 px, 59.997002 Hz
    3840x2160 px, 23.976000 Hz
    3840x2160 px, 24.000000 Hz
    3840x2160 px, 29.969999 Hz
    3840x2160 px, 30.000000 Hz
    3840x2160 px, 30.000000 Hz
    3840x2160 px, 59.997002 Hz (preferred, current)
  Position: 0,0
  Transform: normal
  Scale: 1.000000
HDMI-A-1 "Unknown HD60 S+ 0x00000000 (HDMI-A-1)"
  Physical size: 600x340 mm
  Enabled: no
  Modes:
    720x400 px, 70.082001 Hz
    640x480 px, 59.939999 Hz
    640x480 px, 59.939999 Hz
    640x480 px, 60.000000 Hz
    640x480 px, 75.000000 Hz
    720x480 px, 59.939999 Hz
    720x480 px, 59.939999 Hz
    720x480 px, 60.000000 Hz
    720x480 px, 60.000000 Hz
    720x576 px, 50.000000 Hz
    720x576 px, 50.000000 Hz
    800x600 px, 60.317001 Hz
    800x600 px, 75.000000 Hz
    800x600 px, 119.972000 Hz
    1024x768 px, 60.004002 Hz
    1024x768 px, 70.069000 Hz
    1024x768 px, 75.028999 Hz
    1024x768 px, 119.988998 Hz
    1280x720 px, 50.000000 Hz
    1280x720 px, 59.939999 Hz
    1280x720 px, 60.000000 Hz
    1280x800 px, 60.000000 Hz
    1440x900 px, 60.000000 Hz
    1280x1024 px, 75.025002 Hz
    1280x1024 px, 119.958000 Hz
    1680x1050 px, 60.000000 Hz
    1600x1200 px, 60.000000 Hz
    1920x1080 px, 23.976000 Hz
    1920x1080 px, 24.000000 Hz
    1920x1080 px, 25.000000 Hz
    1920x1080 px, 29.969999 Hz
    1920x1080 px, 30.000000 Hz
    1920x1080 px, 50.000000 Hz
    1920x1080 px, 59.939999 Hz
    1920x1080 px, 60.000000 Hz
    1920x1200 px, 60.000000 Hz
    2560x1440 px, 59.951000 Hz
    3840x2160 px, 23.976000 Hz
    3840x2160 px, 24.000000 Hz
    3840x2160 px, 25.000000 Hz
    3840x2160 px, 29.969999 Hz
    3840x2160 px, 30.000000 Hz
    3840x2160 px, 50.000000 Hz
    3840x2160 px, 59.939999 Hz
    3840x2160 px, 60.000000 Hz
    4096x2160 px, 23.976000 Hz
    4096x2160 px, 24.000000 Hz
    3840x2160 px, 60.000000 Hz (preferred)
````

**Comment by TRPB** [comment]
I thought I'd experiment with this a little. The capture card runs at 60000mHz while the monitors run at 59997mHz. I figured it could be mismatched displays. However, I've tried the following:

1. DP-3 running at 1920x1080@60000 with the capture card mirroring it. Both the capture card and monitor support this mode: Resulted in it showing a grey background and freezing. So it got further but was frozen before it even loaded wf-background or my panel.
2. DP-2 configured to mirror DP-3, HDMI-A-1 off. If the problem with HDMI-A-1 mirroring DP-3 is an issue with slightly mismatched modes then this should work as they are identical monitors: Wayfire didn't segfault but I got nothing on DP-3 and full screen graphical corruption on DP-2 and it was totally frozen.

HDMI-A-1 does work fine if I configure it as a third display next to one of the others.
